app/stores/data-source.ts

```diff
@@ -14,8 +14,10 @@ type DataSourceStore = {
   setDataSource: (value: string) => void;
 };
 
+const STORAGE_KEY = "dataSource";
+
 const customPersist =
-  (name: string, config: StateCreator<DataSourceStore>) =>
+  (config: StateCreator<DataSourceStore>) =>
```

Comment:
Yes I think it's better not to have the name here since in fact the persist is not generic.